src/generators/java/JavaGenerator.ts

```diff
@@ -48,4 +50,46 @@ export class JavaGenerator extends AbstractGenerator<JavaOptions> {
     const result = await renderer.runSelfPreset();
     return RenderOutput.toRenderOutput({result, dependencies: renderer.dependencies});
   }
+
+  /**
+   * Generates all the models to an output directory. 
+   * 
+   * This function is invasive, as it overwrite any existing files with the same name as the model.
+   * 
+   * @param input
+   * @param outputDirectory where you want the models generated to
+   * @param packageName for the models
+   */
+  public async generateToFile(input: Record<string, unknown> | CommonInputModel, outputDirectory: string, packageName: string): Promise<OutputModel[]> {
+    const generatedModels = await this.generateFullOutput(input, packageName);
+    for (const outputModel of generatedModels) {
+      const outputFilePath = path.resolve(outputDirectory, `${FormatHelpers.toPascalCase(outputModel.model.$id || 'undefined')}.java`);
```

Comment:
Like in another comment - it can produce something like `undefined.java` and you can end with several undefined models, so please give me one case where such a situation will be possible. If not, please filter undefined models.